### ISSUE TITLE
WpfPlot fails to load in Visual Studio when using .NET Framework

### ISSUE THREAD

**Issue by bhairav-thakkar** [swharden/ScottPlot#473]
**Description**
Upon trying to reproduce the plot as explained in CookBook, error is encountered when a wpfPlot control is added to the window xaml. The message reads: "Could not load file or assembly 'ScottPlot.WPF version 4.0.35, Culture=neutral' or one of its dependencies. The system can not find the file specified.". 

This is a design-time error. When we run the program, it runs without errors,.  While the program runs, the reporting of error in designer is annoying.

**To Reproduce**

Create a new visual studio wpf project. I used Visual Studio 2019 with .Net Framework 4.7.2. Add the control. Visual studio shows squiggly blue lines and you see the above mentioned error in tool tip text.
 
 

**Screenshots**
If applicable, add screenshots to help explain your problem

At design time, the control shows:
![image](https://user-images.githubusercontent.com/32364304/85035607-60a33f00-b151-11ea-9d59-c891ef1f82f2.png)

The error helper:

![image](https://user-images.githubusercontent.com/32364304/85035885-a7913480-b151-11ea-8bc4-86fe598d028f.png)

**Comment by swharden** [comment]
Hey @bhairav-thakkar, thanks for reporting this! 

I think this is the same issue as #466 and #356 - see those for the solution (updating your csproj format).

I'll research this topic more and see if I can figure out how to prevent this issue in the future

**Comment by bhairav-thakkar** [comment]
Hey @swharden , thank you so much for the super prompt reply. I followed the solution proposed on #356, but I still see the squiggly lines and the designer does not show the wpfplot. 

 
![image](https://user-images.githubusercontent.com/32364304/85046600-a74b6600-b15e-11ea-9dad-9bed87cdac1b.png)

**Comment by swharden** [comment]
Just to confirm, did you try making a new .NET Core WPF project and adding ScottPlot? Does it work then?

... then you modified the .csproj file and edited `TargetFramework` to read `net472`, does it work then?

Learning more about which step fails will be helpful.

**Comment by bhairav-thakkar** [comment]
No, I made a new .NET **Framework** project and added ScottPlot. In the csproj file, if I change the TargerFramework from v4.7.2 to net472, the project breaks with a message -out of range.

**Comment by swharden** [comment]
> No, I made a new .NET **Framework** project and added ScottPlot. 

I'm pretty sure this is the problem.

Create a .NET **Core** project, then modify the csproj file so it builds targeting .NET **Framework**.

The difference is that the .NET Core new project wizard creates SDK-style csproj files, and the .NET Framework new project wizard does not. I still don't know why, but it seems the WpfPlot requires SDK-style csproj files to function at design time.

**Comment by bhairav-thakkar** [comment]
Oh ok. Got it. Thanks. I will check that out. I was wondering what should be the best approach to include the wpfplot in an existing .Net framework project.

**Comment by bhairav-thakkar** [comment]
Hi @swharden , I find the same issue even if I create a new core project and change the targetframework. Attaching the file for your reference.
[WpfApp2.zip](https://github.com/swharden/ScottPlot/files/4800677/WpfApp2.zip)

**Comment by swharden** [comment]
Thanks for sending that ZIP! It indeed fails on my computer, but I took a very close look at it and am surprised I can't spot the difference.

When I create a new project from scratch and do these steps, it works!
* create a new .NET Core WPF project
* manage nuget packages
* install `ScottPlot.WPF`
* drag/drop WpfPlot from the Toolbox onto the Window
* rebuild (verify it works)
* edit csproj to target `net472`
* rebuild (still works)

Here's a zip of that project
[WpfApp3.zip](https://github.com/swharden/ScottPlot/files/4801234/WpfApp3.zip)

![image](https://user-images.githubusercontent.com/4165489/85073001-ff4a9280-b187-11ea-8a2d-c88b63ea2457.png)

**Comment by riquich** [comment]
hello Scott, I am having the same problem aswell, downloaded the zip you provided, opened it and the problem still exists. Functionally it has no effect but seeing the error message all the time might become frustrating over time. Below is the screenshot.

![image](https://user-images.githubusercontent.com/34640902/85224399-d23df000-b3d2-11ea-9716-8ea5dbf49b9c.png)

Thank you for your interest

**Comment by swharden** [comment]
> hello Scott, I am having the same problem aswell, downloaded the zip you provided, opened it and the problem still exists. Functionally it has no effect but seeing the error message all the time might become frustrating over time. 

@riquich Thanks for confirming you have this same issue! I just reformatted and installed a fresh Windows 10 and latest Visual Studio, and I now too get the same error message. Here are those details in a searchable format:

```
Exception: Cannot create an instance of "WpfPlot".
 at Microsoft.VisualStudio.DesignTools.Designer.InstanceBuilders.InstanceBuilderOperations.InstantiateType(Type type, Boolean supportInternal)
   at Microsoft.VisualStudio.DesignTools.Designer.InstanceBuilders.ClrObjectInstanceBuilder.InstantiateTargetType(ILocalInstanceBuilderContext context, ViewNode viewNode)
   at Microsoft.VisualStudio.DesignTools.Designer.InstanceBuilders.ClrObjectInstanceBuilder.Instantiate(ILocalInstanceBuilderContext context, ViewNode viewNode)
   at Microsoft.VisualStudio.DesignTools.WpfDesigner.InstanceBuilders.FrameworkElementInstanceBuilder.Instantiate(ILocalInstanceBuilderContext context, ViewNode viewNode)
   at Microsoft.VisualStudio.DesignTools.WpfDesigner.InstanceBuilders.UserControlInstanceBuilder.Instantiate(ILocalInstanceBuilderContext context, ViewNode viewNode)
   at Microsoft.VisualStudio.DesignTools.Designer.InstanceBuilders.LocalInstanceManager.CreateInstance(IInstanceBuilder builder, ViewNode viewNode)
```

```
TargetInvocationException: Exception has been thrown by the target of an invocation.
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic)
```

```
FileNotFoundException: Could not load file or assembly 'ScottPlot.WPF, Version=4.0.35, Culture=neutral' or one of its dependencies. The system cannot find the file specified.
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at System.Windows.Navigation.BaseUriHelper.GetLoadedAssembly(String assemblyName, String assemblyVersion, String assemblyKey)
   at MS.Internal.AppModel.ResourceContainer.GetResourceManagerWrapper(Uri uri, String& partName, Boolean& isContentFile)
   at MS.Internal.AppModel.ResourceContainer.GetPartCore(Uri uri)
   at System.IO.Packaging.Package.GetPartHelper(Uri partUri)
   at System.IO.Packaging.Package.GetPart(Uri partUri)
   at System.Windows.Application.GetResourceOrContentPart(Uri uri)
   at System.Windows.Application.LoadComponent(Object component, Uri resourceLocator)
   at ScottPlot.WpfPlot.InitializeComponent()
   at ScottPlot.WpfPlot..ctor()
```

![image](https://user-images.githubusercontent.com/4165489/85233804-fe02ab00-b3bd-11ea-9c50-5e4f7fe4c28d.png)

**Comment by swharden** [comment]
I'm thinking this relates to how Visual Studio (which is 32-bit) caches DLLs, and 32-bit vs. 64-bit DLLs may not be agreeing with each other. An important note is that ***this is an error with Visual Studio designer***, not with ScottPlot or the build system. As we've seen, the built application works fine.

A solution _to stop displaying a design-time error message_ is to modify your project to target x64. It seems when this is done Visual Studio doesn't even attempt to display the control.

* create a new .NET Framework 4.7.1 project
* modify the build configuration for `All Configurations` and set platform target to `x64`
* Install ScottPlot.WPF package with NuGet
* rebuild the project
* drag/drop a WpfPlot from the toolbox onto the window (it will fail to display)
* close Visual Studio and open it again (it won't display an error anymore)

![image](https://user-images.githubusercontent.com/4165489/85235860-754c3500-b3e6-11ea-9764-e196cb4aac00.png)

![image](https://user-images.githubusercontent.com/4165489/85235897-df64da00-b3e6-11ea-81cd-a17df8f39518.png)

I'll keep investigating if there's something I can do to prevent this error in the future, because users shouldn't have to know how to do all this 😝

**Comment by riquich** [comment]
Hi Scott, thank you for the tip, it worked and removed the error !

**Comment by swharden** [comment]
> Hi Scott, thank you for the tip, it worked and removed the error !

I'm glad you got it working! It's crazy how many hours this sucked up today, but I think I narrowed down the issue further: resizing a canvas in a WPF user control crashes the control in Visual Studio's designer view for .NET Framework (but not .NET Core) applications. What a mouthful! f06cfe316e92448d7fce1a4e1fccdaabc0a9d6a4 may fix the problem entirely. I'll publish this on NuGet shortly and update here when I do...

**Comment by swharden** [comment]
Ah man so close! That trick seemed to work locally, but not when it's on NuGet. I just released ScottPlot 4.0.36 and this issue persists. I think it may have to do with defining the `AssemblyVersion`, but I'll have to research this further...

https://github.com/swharden/ScottPlot/blob/caf440c3a02464ca592c66ab2d1fde38ae58d0f3/src/ScottPlot.WPF/ScottPlot.WPF.csproj#L21

**Comment by riquich** [comment]
Hi Scott, thanks for the endeavor, hope you find the solution soon, without much effort

**Comment by swharden** [comment]
Yay this is finally fixed in ScottPlot 4.0.37 🎉

Solution: `AssemblyVersion` must be in `x.x.x.x` format (cannot be `x.x.x` format) for .NET Framework, even though .NET Core doesn't seem to care.
